src/patch-events.js

```diff
@@ -381,9 +381,16 @@ export function addEventListener(type, fnOrObj, optionsOrCapture) {
         }
         return;
       }
-      // prevent non-bubbling events from triggering bubbling handlers on shadowroot, but only if not in capture phase
-      if (e.eventPhase !== Event.CAPTURING_PHASE && !e.bubbles && e.target !== target && !(target instanceof Window)) {
-        return;
+      if (e.eventPhase !== Event.CAPTURING_PHASE && e.target !== target && !(target instanceof Window)) {
+        // prevent non-bubbling events from triggering bubbling handlers on shadowroot, but only if not in capture phase
+        if (!e.bubbles) {
+          return;
+        }
+        // if the bubbling event is dispatched on the node that has the shadowRoot
```

Comment:
Is this the only case that an event reach a host without reaching its `shadowRoot`? If so, maybe this is ok, but we should put an explicit comment. If not, then do we need to check if the `shadowRoot` is in the composed path? 

We probably want some additional comments here at the least since it's important to remember that it's only possible for this to happen because when we listen on `shadowRoots` we actually listen on its host, right?

Reply:
Also, shouldn't this event be prevented if we're in the capture also? We'll probably need a test for that case.

Reply:
Upon further reflection with @azakus, you cannot observe this problem except in this case right now. However, this is due to a limitation of the polyfill in dealing with undistributed nodes. If the polyfill correctly propagated events dispatched on undistributed nodes, the code here would be inadequate. Consider:

```html
<host> <!-- polyfilled event listener actually added on this node -->
   #shadowRoot <!-- event listener added on this node -->
  <child> <!-- bubbling event dispatched here -->
```

In that case, we'd need to check if the `#shadowRoot` was in the event path and stop the event if not.

We may decide to put undistributed nodes *in* the dom tree hidden to address issues like the event propagation one described above. If we do, then we'll need to modify this code. We may want to do this now to be more robust to that future change.

Reply:
Done